molgenis-security/src/main/java/org/molgenis/security/twofactor/auth/TwoFactorAuthenticationFilter.java

```diff
@@ -49,8 +49,7 @@ protected void doFilterInternal(HttpServletRequest httpServletRequest, HttpServl
 			if (!httpServletRequest.getRequestURI().contains(TwoFactorAuthenticationController.URI)
 					&& SecurityUtils.currentUserIsAuthenticated() && !httpServletRequest.getRequestURI()
 																						.toLowerCase()
-																						.endsWith(
-																								CHANGE_PASSWORD_URI.toLowerCase()))
+																						.equals(CHANGE_PASSWORD_URI.toLowerCase()))
```

Comment:
Since this is security stuff I would like to see a unit test for this code branch.